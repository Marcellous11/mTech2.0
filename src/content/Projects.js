import React from 'react';
import { TypeAnimation } from 'react-type-animation';
import '../styles/Projects.css';
import mTechgif from '../media/mTechGIF1.0.gif';
import leadersGIF from '../media/localLeadersGIF.gif';
import joblyGIF from '../media/joblyGIF.gif';
import yodlrGIF from '../media/YodlrGIF.gif';
import mTech2 from '../media/mTech2.0copy.gif';
const Projects = () => {
	return (
		<div className="Projects ">
			<p className="Title">Projects</p>
			<ul className="Projects-Mods">
				<li>
					<TypeAnimation cursor={false} sequence={[ 'First personal website' ]} />
					<a href="https://github.com/Marcellous11/marcellous.tech" target={'_blank'}>
						<img src={mTechgif} alt="A gif of my website" />
					</a>
				</li>
				<li>
					<TypeAnimation cursor={false} sequence={[ 'Website that shows local government officials' ]} />
					<a href="https://github.com/Marcellous11/full-local-leaders.git" target={'_blank'}>
						<img src={leadersGIF} alt="Gif of local leaders website" />
					</a>
				</li>
				<li>
					<TypeAnimation cursor={false} sequence={[ 'Jobly - Frontend Project' ]} />
					<a href="https://github.com/Marcellous11/React-Jobly/edit/main/README.md" target={'_blank'}>
						<img src={joblyGIF} alt="GIF of Jobly frontend" />
					</a>
				</li>
				<li>
					<TypeAnimation cursor={false} sequence={[ 'Yodlr - Frontend Project' ]} />
					<a href="https://github.com/Marcellous11/yolder_front_end/blob/main/README.md" target={'_blank'}>
						<img src={yodlrGIF} alt="GIF of Yodlr frontend" />
					</a>
				</li>
				<li>
					<TypeAnimation cursor={false} sequence={[ 'mTech2.0 - (this website!) Frontend' ]} />
					<a href="https://github.com/Marcellous11/mTech2.0.git" target={'_blank'}>
						<img src={mTech2} alt="GIF of Yodlr frontend" />
					</a>
				</li>
			</ul>
		</div>
	);
};
export default Projects;
