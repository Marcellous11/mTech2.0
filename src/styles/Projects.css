.Projects-Mods li {
	margin: 2em 0;
	list-style-type: none;
	display: flex;
	flex-direction: column;
	margin-right: 2em;
}
.Projects-Mods li span {
	color: rgb(255, 86, 227);
	font-size: 1em;
	font-weight: 600;
}

.Projects-Mods li:nth-of-type(even) {
	/* margin-left: 5rem; */
	/* flex-direction: row-reverse; */
	margin-right: 1em;
}
.Projects-Mods li div {
	color: red;
}
.Projects-Mods li a img {
	border-radius: .2em;
	margin-right: 2em;
	width: 400px;
}

@media only screen and (max-width: 830px) {
	.Projects-Mods li {
		margin-left: -2em;
	}
	.Projects-Mods li a img {
		width: 250px;
	}
}
