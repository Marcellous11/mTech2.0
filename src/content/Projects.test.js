import { logRoles, prettyDOM, render, screen } from '@testing-library/react';
import Projects from './Projects';

test('Smoke test, does the component render', () => {
	render(<Projects />);
});

test('Snapshot ', () => {
	const { asFragment } = render(<Projects />);

	expect(asFragment()).toMatchSnapshot();
});

test('should have "A gif of my website" in alt text', () => {
	const { getByText, getAllByText, getByAltText } = render(<Projects />);

	expect(getByAltText(`A gif of my website`)).toBeInTheDocument();
});
test('should have "Gif of local leaders website" in alt text', () => {
	const { getByText, getAllByText, getByAltText } = render(<Projects />);

	expect(getByAltText(`Gif of local leaders website`)).toBeInTheDocument();
});
test('should have "GIF of Jobly frontend" in alt text', () => {
	const { getByText, getAllByText, getByAltText } = render(<Projects />);

	expect(getByAltText(`GIF of Jobly frontend`)).toBeInTheDocument();
});
test('should have "GIF of Yodlr frontend" in alt text', () => {
	const { getByText, getAllByText, getByAltText } = render(<Projects />);

	expect(getByAltText(`GIF of Yodlr frontend`)).toBeInTheDocument();
});
