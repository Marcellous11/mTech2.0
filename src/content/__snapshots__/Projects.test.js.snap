// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot  1`] = `
<DocumentFragment>
  <div
    class="Projects "
  >
    <p
      class="Title"
    >
      Projects
    </p>
    <ul
      class="Projects-Mods"
    >
      <li>
        <span
          class=""
        />
        <a
          href="https://github.com/Marcellous11/marcellous.tech"
          target="_blank"
        >
          <img
            alt="A gif of my website"
            src="mTechGIF1.0.gif"
          />
        </a>
      </li>
      <li>
        <span
          class=""
        />
        <a
          href="https://github.com/Marcellous11/full-local-leaders.git"
          target="_blank"
        >
          <img
            alt="Gif of local leaders website"
            src="localLeadersGIF.gif"
          />
        </a>
      </li>
      <li>
        <span
          class=""
        />
        <a
          href="https://github.com/Marcellous11/React-Jobly/edit/main/README.md"
          target="_blank"
        >
          <img
            alt="GIF of Jobly frontend"
            src="joblyGIF.gif"
          />
        </a>
      </li>
      <li>
        <span
          class=""
        />
        <a
          href="https://github.com/Marcellous11/yolder_front_end/blob/main/README.md"
          target="_blank"
        >
          <img
            alt="GIF of Yodlr frontend"
            src="YodlrGIF.gif"
          />
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
